sync-pipelines/src/commonMain/kotlin/com/quran/shared/pipeline/SyncEnginePipeline.kt

```diff
@@ -0,0 +1,123 @@
+package com.quran.shared.pipeline
+
+import co.touchlab.kermit.Logger
+import com.quran.shared.mutations.LocalModelMutation
+import com.quran.shared.mutations.RemoteModelMutation
+import com.quran.shared.persistence.repository.PageBookmarksSynchronizationRepository
+import com.quran.shared.syncengine.AuthenticationDataFetcher
+import com.quran.shared.syncengine.LocalModificationDateFetcher
+import com.quran.shared.syncengine.LocalMutationsFetcher
+import com.quran.shared.syncengine.PageBookmark
+import com.quran.shared.syncengine.PageBookmarksSynchronizationConfigurations
+import com.quran.shared.syncengine.ResultNotifier
+import com.quran.shared.syncengine.SynchronizationClient
+import com.quran.shared.syncengine.SynchronizationClientBuilder
+import com.quran.shared.syncengine.SynchronizationEnvironment
+
+interface SyncEngineCallback {
+    fun synchronizationDone(newLastModificationDate: Long)
+    fun encounteredError(errorMsg: String)
+}
+
+public class SyncEnginePipeline(
+    val bookmarksRepository: PageBookmarksSynchronizationRepository
+) {
+    var syncClient: SynchronizationClient? = null
```

Comment:
nit -

```kotlin
latent var syncClient: SynchronizationClient
```

this way you can use it without making it `nullable` since you're sure `setup` is required first. alternatively / probably preferably, we can just move them to the constructor.